src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateMethodSymbol.cs

```diff
@@ -44,7 +44,10 @@ protected void InitializeParameters(ImmutableArray<ParameterSymbol> parameters)
             BindingDiagnosticBag diagnostics)
         {
             Binder binder = delegateType.GetBinder(syntax.ParameterList);
-            TypeSyntax returnTypeSyntax = syntax.ReturnType.SkipRef(out RefKind refKind);
+            TypeSyntax returnTypeSyntax = syntax.ReturnType;
+            Debug.Assert(returnTypeSyntax is not ScopedTypeSyntax);
+
+            returnTypeSyntax = returnTypeSyntax.SkipScoped(out _).SkipRef(out RefKind refKind);
```

Comment:
We have this pattern repeated many times. Consider creating a `SkipUnexpectedScoped(this TypeSyntax syntax)` helper method.

Reply:
> Consider creating a ```SkipUnexpectedScoped(this TypeSyntax syntax)``` helper method.

I see close to no benefit in adding a helper like that. Even if ```scoped``` was supported here, we would have to skip it anyway.